pkg/tsdb/cloudwatch/models/settings_test.go

```diff
@@ -71,4 +72,39 @@ func Test_Settings_LoadCloudWatchSettings(t *testing.T) {
 		assert.Equal(t, "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY", s.SecretKey)
 		assert.Equal(t, "AKIAIOSFODNN7EXAMPLE", s.AccessKey)
 	})
+	t.Run("Should set logsTimeout to default duration if it is not defined", func(t *testing.T) {
+		settings := backend.DataSourceInstanceSettings{
+			ID: 33,
+			JSONData: []byte(`{
+			"authType": "arn",
+			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana"
+		  }`),
+			DecryptedSecureJSONData: map[string]string{
+				"accessKey": "AKIAIOSFODNN7EXAMPLE",
+				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+			},
+		}
+
+		s, err := LoadCloudWatchSettings(settings)
+		require.NoError(t, err)
+		assert.Equal(t, Duration(time.Minute * 30), s.LogsTimeout)
+	})
+	t.Run("Should correctly parse logsTimeout json field", func(t *testing.T) {
+		settings := backend.DataSourceInstanceSettings{
+			ID: 33,
+			JSONData: []byte(`{
+			"authType": "arn",
+			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana",
+			"logsTimeout": "10m"
+		  }`),
+			DecryptedSecureJSONData: map[string]string{
+				"accessKey": "AKIAIOSFODNN7EXAMPLE",
+				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+			},
+		}
+
+		s, err := LoadCloudWatchSettings(settings)
+		require.NoError(t, err)
+		assert.Equal(t, Duration(time.Minute * 10), s.LogsTimeout)
+	})
```

Comment:
I would add all the possible test cases here to document our behavior and to enter all possible paths.
(Here's what I was using to test out my other comment to you setting the default in error cases if this is a good starting point)
```golang
func Test_LoadCloudWatchSettings_logsTimeout(t *testing.T) {
	t.Run("Should set logsTimeout to default duration if it is not defined", func(t *testing.T) {
		settings := backend.DataSourceInstanceSettings{
			ID: 33,
			JSONData: []byte(`{
			"authType": "arn",
			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana"
		  }`),
			DecryptedSecureJSONData: map[string]string{
				"accessKey": "AKIAIOSFODNN7EXAMPLE",
				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
			},
		}

		s, err := LoadCloudWatchSettings(settings)
		require.NoError(t, err)
		assert.Equal(t, 30*time.Minute, s.LogsTimeout.Duration)
	})
	t.Run("Should parse valid logsTimeout json field duration string", func(t *testing.T) {
		settings := backend.DataSourceInstanceSettings{
			ID: 33,
			JSONData: []byte(`{
			"authType": "arn",
			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana",
			"logsTimeout": "10m"
		  }`),
			DecryptedSecureJSONData: map[string]string{
				"accessKey": "AKIAIOSFODNN7EXAMPLE",
				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
			},
		}

		s, err := LoadCloudWatchSettings(settings)
		require.NoError(t, err)
		assert.Equal(t, 10*time.Minute, s.LogsTimeout.Duration)
	})

	t.Run("Should return default timeout for an invalid duration string", func(t *testing.T) {
		settings := backend.DataSourceInstanceSettings{
			ID: 33,
			JSONData: []byte(`{
			"authType": "arn",
			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana",
			"logsTimeout": "bad value"
		  }`),
			DecryptedSecureJSONData: map[string]string{
				"accessKey": "AKIAIOSFODNN7EXAMPLE",
				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
			},
		}

		s, err := LoadCloudWatchSettings(settings)
		require.NoError(t, err)
		assert.Equal(t, 30*time.Minute, s.LogsTimeout.Duration)
	})

	t.Run("Should return default timeout for logsTimeout json field of unexpected type", func(t *testing.T) {
		settings := backend.DataSourceInstanceSettings{
			ID: 33,
			JSONData: []byte(`{
			"authType": "arn",
			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana",
			"logsTimeout": true
		  }`),
			DecryptedSecureJSONData: map[string]string{
				"accessKey": "AKIAIOSFODNN7EXAMPLE",
				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
			},
		}

		s, err := LoadCloudWatchSettings(settings)
		require.NoError(t, err)
		assert.Equal(t, 30*time.Minute, s.LogsTimeout.Duration)
	})

	t.Run("Should correctly parse logsTimeout json field duration in nanoseconds", func(t *testing.T) {
		settings := backend.DataSourceInstanceSettings{
			ID: 33,
			JSONData: []byte(`{
			"authType": "arn",
			"assumeRoleArn": "arn:aws:iam::123456789012:role/grafana",
			"logsTimeout": 300000000
		  }`),
			DecryptedSecureJSONData: map[string]string{
				"accessKey": "AKIAIOSFODNN7EXAMPLE",
				"secretKey": "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
			},
		}

		s, err := LoadCloudWatchSettings(settings)
		require.NoError(t, err)
		assert.Equal(t, 300*time.Millisecond, s.LogsTimeout.Duration)
	})
}
```
(I don't think we can actually reach the json unmarshal error case in our custom Unmarshaller... I don't think we can't write a test for it. This happens sometimes, but it's sort of like trying to test the `json` module in the standard library and I think there is no need since that is expected to be quite reliable already.)